pyinfra/facts/server.py

```diff
@@ -796,3 +796,52 @@ def process(self, output):
             )
 
         return limits
+
+
+class RebootRequired(FactBase[bool]):
+    """
+    Returns a boolean indicating whether the system requires a reboot.
+
+    On Linux systems:
+    - Checks /var/run/reboot-required and /var/run/reboot-required.pkgs
+    - On Alpine Linux, compares installed kernel with running kernel
+
+    On FreeBSD systems:
+    - Compares running kernel version with installed kernel version
+    """
+
+    def command(self) -> str:
+        return """
+# Get OS type
+OS_TYPE=$(uname -s)
```

Comment:
Ah yes, the problem of facts being unable to call each other. Unfortunately I think this is the best way we can currently do this.